edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesRuntimeInfoProviderTest.cs

```diff
@@ -173,48 +173,47 @@ public async Task ReturnsLastKnowModuleState()
 
             Dictionary<string, V1Pod> modified = BuildPodList();
 
-            DateTime agentStartTime = new DateTime(2019, 6, 11);
+            string agentWaitingReason = "CrashBackLoopOff";
```

Comment:
Could you possibly split this monstrous test in several once? This test violates one of the most important principles: The [Single Responsibility](https://en.wikipedia.org/wiki/Single_responsibility_principle) principle. What is the main thing that verifies this test? We can discuss it offline.